device/samples/pnp/pnpTemperatureController.js

```diff
@@ -97,30 +98,41 @@ const updateComponentReportedProperties = (deviceTwin, patch, componentName) =>
   });
 };
 
-const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin) => {
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin, componentNames) => {
   deviceTwin.on('properties.desired', (delta) => {
-    if (delta[thermostat1ComponentName] || delta[thermostat2ComponentName] || delta[deviceInfoComponentName]) {
+    console.log('Received an update for device with value: ' + JSON.stringify(delta));
+    Object.entries(delta).forEach(([key, values]) => {
       const version = delta.$version;
-      const componentName = Object.keys(delta)[0];
-      let logLine;
-      if (!!(componentName)) {
-        logLine = 'Received an update for component: ' + componentName + ' with value: ' + JSON.stringify(delta);
-      } else {
-        logLine = 'Received an update for root interface with value: ' + JSON.stringify(delta);
+      if (!!(componentNames) && componentNames.includes(key)) { // then it is a component we are expecting
+        const componentName = key;
+        const patchForComponents = { [componentName]: {} };
+        Object.entries(values).forEach(([propertyName, propertyValue]) => {
+          if (propertyName !== '__t' && propertyName !== '$version') {
+            console.log('Will update property: ' + propertyName + ' to value: ' + propertyValue + ' of component: ' + componentName);
+            const propertyContent = { value: propertyValue };
+            propertyContent.ac = 200;
+            propertyContent.ad = 'Successfully executed patch';
+            propertyContent.av = version;
+            patchForComponents[componentName][propertyName] = propertyContent;
+          }
+        });
+        updateComponentReportedProperties(deviceTwin, patchForComponents, componentName);
       }
-      console.log(logLine);
-      const patch = { [componentName]: {} };
-      Object.entries(delta[componentName]).forEach(([propertyName, propertyValue]) => {
-        if (propertyName !== '__t') {
-          const propertyContent = { value: propertyValue };
-          propertyContent.ac = 200;
-          propertyContent.ad = 'Successfully executed patch';
-          propertyContent.av = version;
-          patch[componentName][propertyName] = propertyContent;
-        }
-      });
-      updateComponentReportedProperties(deviceTwin, patch, componentName);
-    }
+      else if  (key !== '$version') {
+        const patchForRoot = { };
+        Object.entries(delta).forEach(([propertyName, propertyValue]) => {
```

Comment:
>Object.entries(delta).forEach(([propertyName, propertyValue]) => { [](start = 8, length = 66)

In addition in reference to my above comment, I believe the reports for a,b, and c could be in error.

the desired looks like below (not showing d):
{
   a: {
         __t: 'c',
         bob: 1,
          carol: 2
       },
    b: {
        __t: 'c',
        ted: 3,
        alice: 4
     }

Could you end up reporting:
    a: {
          value: {__t: 'c', bob: 1, carol: 2},
          ac: 200,
          av: 2
        },
     b: {
          value: {__t: 'c', ted:  3, alice: 4},
          ac: 200,
           av: 2
        }